File: 프로그래머스/0/181846. 두 수의 합/두 수의 합.java
class Solution {
    public String solution(String a, String b) {
        int n = Math.max(a.length(), b.length());
        char[] arr = new StringBuilder(a).reverse().toString().toCharArray();
        char[] brr = new StringBuilder(b).reverse().toString().toCharArray();
        int prev = 0;
        StringBuilder ret = new StringBuilder();
        for(int i = 0; i < n; i++){
            int aNum = i + 1 > arr.length ? 0 : arr[i] - '0';
            int bNum = i + 1 > brr.length ? 0 : brr[i] - '0';
            int sum = aNum + bNum + prev;
            prev = 0;
            if(sum >= 10){prev = 1; sum -= 10;}
            ret.append(sum);
        }
        if(prev == 1) ret.append(1);
        return ret.reverse().toString();
    }
}

File: 프로그래머스/0/181846. 두 수의 합/두 수의 합.py
def solution(a, b):
    a, b = a[::-1], b[::-1]
    ssum = 0
    idx = 0
    ret = ''
    prev = 0
    print(a, b)
    while idx < max(len(a), len(b)):
        na = int(a[idx]) if idx < len(a) else 0
        nb = int(b[idx]) if idx < len(b) else 0
        ssum = na + nb + prev
        if ssum >= 10:
            prev, ssum = 1, ssum-10
        else:
            prev = 0
        ret += str(ssum)
        idx += 1
        
    if prev == 1: ret += "1"
    return ret[::-1]

# 2015793246801012701479981
# 287346502836570928366
# 305793246910280479981